waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingService.java

```diff
@@ -181,9 +182,12 @@ private boolean includeInResults(MetaStoreMapping metaStoreMapping, String prefi
   @Override
   public DatabaseMapping databaseMapping(@NotNull String databaseName) {
     // Find a Metastore with a prefix
-    for (String metastorePrefix : mappingsByPrefix.keySet()) {
+    Iterator<Entry<String, DatabaseMapping>> iterator = mappingsByPrefix.entrySet().iterator();
```

Comment:
I find the left side much easier to read imo

Reply:
I think the way it was before is more readable, so I think this is mainly a question of whether we'd rather change the code to eliminate the warnings or silence the warnings because we don't care about them.

Reply:
It is more readable but less efficient, not that the efficiency gains in this case will make any real difference. I  do think that the general pattern of iterating over the entries rather than the keys when you use both the keys and values is preferable though.